src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -598,6 +581,13 @@ private BoundExpression BindExpressionInternal(ExpressionSyntax node, BindingDia
 
             BoundExpression result = bindExpressionInternal(node, diagnostics, invoked, indexed);
 
+            // Only invocation expressions need to resolve extension members as invoked (ie. filtering out non-invocable members)
+            // All other contexts can go ahead and resolve to a non-method extension member
+            if (!invoked)
+            {
+                result = ResolveToNonMethodExtensionMemberIfPossible(result, diagnostics);
```

Comment:
>ResolveToNonMethodExtensionMemberIfPossible [](http://example.com/codeflow?start=25&length=43)

Is the name somewhat confusing? We may still resolve to methods, right?

Reply:
Maybe I didn't quite understand the question. `ResolveToNonMethodExtensionMemberIfPossible` only returns non-methods.